cnd/src/herc20.rs

```diff
@@ -30,7 +30,7 @@ pub async fn new<C>(
     role: Role,
     side: Side,
     states: Arc<States>,
-    connector: Arc<C>,
+    connector: impl AsRef<C>,
```

Comment:
Yes, less `Arc` and more `impl As Ref` in the codebase would be good actually. It would make the APIs more stable I think.

Reply:
I think so too yes, only depend on what you need :)